tests/test_core.py

```diff
@@ -1280,8 +1280,10 @@ def test_exceptions_allowed_2(self):
     # mismatches are fixed in fixImports() function in JS glue code.
     self.set_setting('EXCEPTION_CATCHING_ALLOWED', ["main"])
     # otherwise it is inlined and not identified
-    self.set_setting('INLINING_LIMIT')
+    self.set_setting('INLINING_LIMIT', 50)
```

Comment:
This was previously set to 1 (`set_setting`'s default argument is 1), but the comment says "otherwise it is inlined and not identified", which didn't make sense, because this comment says we were effectively preventing inlining altogether while `INLINING_LIMIT = 1` means minimizing the inlining threshold.

Turned out I somehow changed `50` to `1` in 2016 while doing other work, I guess by mistake: https://github.com/emscripten-core/emscripten/pull/4489/files#diff-938de46e643ab091cff6a7c23e4e752e8907e2266f47889d988923352f7a1058R1484
But it seems luckily `nocatch` function in the test haven't been inlined so far; otherwise the test wouldn't have succeeded. Anyway I'm reverting this to the original `50`, which was our intention that we wanted to prevent inlining.

Reply:
I think the specific value no longer matters. A long time ago we used to use `INLINING_LIMIT` to control inlining in JS by closure compiler (or maybe elsewhere in the JS optimizer), but it looks like we no longer do. Perhaps that was removed with fastcomp. In any case, right now `INLINING_LIMIT` is effectively a boolean (that controls adding `-fno-inline-functions`) AFAICT.

Reply:
Thanks. Then I think the current explanation in the settings, which is misleading: https://github.com/emscripten-core/emscripten/blob/42cd632ebfc187eeab263baeea49fa011f14b530/src/settings.js#L293-L298

After reading it, I thought setting it to 1 would allow inlining to the maximum extent but the current status is it does the opposite. 

Do you think we should change the doc too? If so, do you think we should make the option boolean? I think it is cleaner, but I'm worried if it is possible to break some users' usage or something. Also similarly for `DISABLE_CATCHING_EXCEPTION`, @sbc100 [suggested](https://github.com/emscripten-core/emscripten/pull/13725#pullrequestreview-617050915) we can eliminate `2` and make it boolean but I was concerned about backward compatibility. What do you think?

Reply:
Good point, yeah, those docs are out of date. Yes, I think we should change the doc.

I think it would also be good to make the option boolean. I think that just means updating the test suite to not write anything but 0 or 1 to it?

Reply:
Yes, I'll land this as is and fix the doc and its value for other tests as a followup.

Reply:
Yes, I'll revert `50` part of the change here and fix the doc (and uses in other tests) as a followup.